+nosnoc/+solver/MpccSolver.m

```diff
@@ -426,5 +458,415 @@ function print_infeasibility(obj, results)
+                    w_init = vertcat(w_orig, G_init, H_init);
+
+                    
+                    casadi_nlp = struct('f', f , 'x', w, 'g', g, 'p', p);
+                    opts_casadi_nlp.ipopt.max_iter = 5000;
+                    opts_casadi_nlp.ipopt.warm_start_init_point = 'yes';
+                    opts_casadi_nlp.ipopt.warm_start_entire_iterate = 'yes';
+                    opts_casadi_nlp.ipopt.warm_start_bound_push = 1e-5;
+                    opts_casadi_nlp.ipopt.warm_start_bound_frac = 1e-5;
+                    opts_casadi_nlp.ipopt.warm_start_mult_bound_push = 1e-5;
+                    opts_casadi_nlp.ipopt.tol = obj.opts.comp_tol;
+                    opts_casadi_nlp.ipopt.acceptable_tol = sqrt(obj.opts.comp_tol);
+                    opts_casadi_nlp.ipopt.acceptable_dual_inf_tol = sqrt(obj.opts.comp_tol);
+                    opts_casadi_nlp.ipopt.fixed_variable_treatment = 'relax_bounds';
+                    opts_casadi_nlp.ipopt.honor_original_bounds = 'yes';
+                    opts_casadi_nlp.ipopt.bound_relax_factor = 1e-12;
+                    opts_casadi_nlp.ipopt.linear_solver = obj.opts.opts_casadi_nlp.ipopt.linear_solver;
+                    opts_casadi_nlp.ipopt.mu_strategy = 'adaptive';
+                    opts_casadi_nlp.ipopt.mu_oracle = 'quality-function';
+                    opts_casadi_nlp.ipopt.nlp_scaling_method = 'none';
+                    default_tol = 1e-7;
+                    opts_casadi_nlp.ipopt.tol = default_tol;
+                    opts_casadi_nlp.ipopt.dual_inf_tol = default_tol;
+                    opts_casadi_nlp.ipopt.dual_inf_tol = default_tol;
+                    opts_casadi_nlp.ipopt.compl_inf_tol = default_tol;
+                    opts_casadi_nlp.ipopt.resto_failure_feasibility_threshold = 0;
+                    opts_casadi_nlp.ipopt.print_level = 5;
+                    opts_casadi_nlp.ipopt.sb = 'yes';
+                    
+                    tnlp_solver = nlpsol('tnlp', 'ipopt', casadi_nlp, opts_casadi_nlp);
+                    tnlp_results = tnlp_solver('x0', w_init,...
+                        'lbx', lbw,...
+                        'ubx', ubw,...
+                        'lbg', lbg,...
+                        'ubg', ubg,...
+                        'lam_g0', lam_g_aug,...
+                        'lam_x0', lam_x_aug,...
+                        'p', nlp.p.mpcc_p().val);
+                    res_out = tnlp_results;
+                    if tnlp_solver.stats.success
+                        converged = true;
+                    else
+                        n_biactive = n_biactive + 1;
+                    end 
+                    %converged = true;
+                end
+                w_star = full(tnlp_results.x);
+                w_star_orig = w_star(ind_mpcc);
+                max(abs(w_init - w_star))
+                [~,sort_idx] = sort(full(abs(w_init - w_star))); % sorted by difference from original stationary point
```

Comment:
for what is this idx used for? why sort the diffrence?

Reply:
This was debugging info. Removing.